core/shared/src/main/scala/fs2/text.scala

```diff
@@ -27,19 +27,39 @@ import java.nio.charset.{
   CharacterCodingException,
   Charset,
   CharsetDecoder,
+  CharsetEncoder,
+  CodingErrorAction,
   MalformedInputException,
   StandardCharsets,
   UnmappableCharacterException
 }
 import scala.collection.mutable.{ArrayBuffer, Builder}
 import scodec.bits.{Bases, ByteVector}
 
+import scala.annotation.tailrec
+
 /** Provides utilities for working with streams of text (e.g., encoding byte streams to strings). */
 object text {
 
+  /** Byte order mark (BOM) values for different Unicode charsets.
+    */
+  object bom {
```

Comment:
As they can be useful downstream (for example, I needed them in http4s-fs2-data) I expose them here, but happy to convert to something private if people think this isn't the right place.

Reply:
The problem is that `Seq[Byte]` is a really inefficient representation. I wonder if we should use a `ByteVector` instead.